frontend/src/components/custom-tools/prompt-card/PromptCardItems.jsx

```diff
@@ -61,18 +80,174 @@ function PromptCardItems({
+              if (a?.isDefault) return -1; // Default profile comes first
+              if (b?.isDefault) return 1;
+              if (a?.isEnabled && !b?.isEnabled) return -1; // Enabled profiles come before disabled
+              if (!a?.isEnabled && b?.isEnabled) return 1;
+              return 0;
+            })
+        );
+      })
+      .catch((err) => {
+        setAlertDetails(handleException(err));
+      });
+  };
+
+  const tooltipContent = (adapterConf) => (
+    <div>
+      {Object.entries(adapterConf)?.map(([key, value]) => (
+        <div key={key}>
+          <strong>{key}:</strong> {value}
+        </div>
+      ))}
+    </div>
+  );
+
+  const handleExpandClick = (profile) => {
+    const profileId = profile?.profile_id;
+    setExpandedProfiles((prevState) =>
+      prevState.includes(profileId)
+        ? prevState.filter((id) => id !== profileId)
+        : [...prevState, profileId]
+    );
+  };
 
+  const handleTagChange = (checked, profileId) => {
+    setEnabledProfiles((prevState) =>
+      checked
+        ? [...prevState, profileId]
+        : prevState.filter((id) => id !== profileId)
+    );
+  };
+
+  const getColSpan = () => (componentWidth < 1200 ? 24 : 6);
+
+  const renderSinglePassResult = () => {
+    const [firstResult] = result || [];
+    if (
+      promptDetails.active &&
+      (firstResult?.output || firstResult?.output === 0)
+    ) {
+      return (
+        <>
+          <Divider className="prompt-card-divider" />
+          <div
+            className={`prompt-card-result prompt-card-div ${
+              expandedProfiles.includes(firstResult.profileManager) &&
+              "prompt-profile-run-expanded"
+            }`}
+          >
+            {isSinglePassExtractLoading ? (
+              <Spin indicator={<SpinnerLoader size="small" />} />
+            ) : (
+              <Typography.Paragraph className="prompt-card-res font-size-12">
+                <div
+                  className={
+                    expandedProfiles.includes(firstResult.profileManager)
+                      ? "expanded-output"
+                      : "collapsed-output"
+                  }
+                >
+                  {displayPromptResult(firstResult.output, true)}
+                </div>
+              </Typography.Paragraph>
+            )}
+            <div className="prompt-profile-run">
+              <Tooltip title="Expand">
+                <Button
+                  size="small"
+                  type="text"
+                  className="prompt-card-action-button"
+                  onClick={() =>
+                    handleExpandClick({
+                      profile_id: firstResult.profileManager,
+                    })
+                  }
+                >
+                  <ArrowsAltOutlined className="prompt-card-actions-head" />
+                </Button>
+              </Tooltip>
+            </div>
+          </div>
+        </>
+      );
+    }
+    return null;
```

Comment:
@jagadeeswaran-zipstack Since we're returning JSX code here, let's return empty fragment (`<></>`) instead of `null`.

Reply:
Got it.